3ds2/src/main/java/com/adyen/checkout/adyen3ds2/DefaultAdyen3DS2Delegate.kt

```diff
@@ -0,0 +1,404 @@
+import com.adyen.checkout.adyen3ds2.exception.Authentication3DS2Exception
+import com.adyen.checkout.adyen3ds2.exception.Cancelled3DS2Exception
+import com.adyen.checkout.adyen3ds2.model.ChallengeToken
+import com.adyen.checkout.adyen3ds2.model.FingerprintToken
+import com.adyen.checkout.adyen3ds2.repository.SubmitFingerprintRepository
+import com.adyen.checkout.adyen3ds2.repository.SubmitFingerprintResult
+import com.adyen.checkout.components.encoding.Base64Encoder
+import com.adyen.checkout.components.model.payments.response.Action
+import com.adyen.checkout.components.model.payments.response.RedirectAction
+import com.adyen.checkout.components.model.payments.response.Threeds2Action
+import com.adyen.checkout.components.model.payments.response.Threeds2ChallengeAction
+import com.adyen.checkout.components.model.payments.response.Threeds2FingerprintAction
+import com.adyen.checkout.core.exception.CheckoutException
+import com.adyen.checkout.core.exception.ComponentException
+import com.adyen.checkout.core.log.LogUtil
+import com.adyen.checkout.core.log.Logger
+import com.adyen.checkout.redirect.handler.RedirectHandler
+import com.adyen.threeds2.AuthenticationRequestParameters
+import com.adyen.threeds2.ChallengeStatusReceiver
+import com.adyen.threeds2.CompletionEvent
+import com.adyen.threeds2.ProtocolErrorEvent
+import com.adyen.threeds2.RuntimeErrorEvent
+import com.adyen.threeds2.ThreeDS2Service
+import com.adyen.threeds2.Transaction
+import com.adyen.threeds2.customization.UiCustomization
+import com.adyen.threeds2.exception.InvalidInputException
+import com.adyen.threeds2.exception.SDKAlreadyInitializedException
+import com.adyen.threeds2.exception.SDKNotInitializedException
+import com.adyen.threeds2.exception.SDKRuntimeException
+import com.adyen.threeds2.parameters.ChallengeParameters
+import com.adyen.threeds2.util.AdyenConfigParameters
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineExceptionHandler
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.channels.BufferOverflow
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.launch
+import org.json.JSONException
+import org.json.JSONObject
+
+@Suppress("TooManyFunctions", "LongParameterList")
+internal class DefaultAdyen3DS2Delegate(
+    private val savedStateHandle: SavedStateHandle,
+    private val configuration: Adyen3DS2Configuration,
+    private val submitFingerprintRepository: SubmitFingerprintRepository,
+    private val adyen3DS2Serializer: Adyen3DS2Serializer,
+    private val redirectHandler: RedirectHandler,
+    private val threeDS2Service: ThreeDS2Service,
+    private val defaultDispatcher: CoroutineDispatcher,
+    private val embeddedRequestorAppUrl: String,
+    private val base64Encoder: Base64Encoder,
+) : Adyen3DS2Delegate, ChallengeStatusReceiver {
+
+    private val _detailsFlow = MutableSharedFlow<JSONObject>(1, 1, BufferOverflow.DROP_OLDEST)
```

Comment:
Can be replaced with `private val _detailsFlow: MutableSharedFlow<JSONObject> = MutableSingleEventSharedFlow()`